test/models/user_test.rb

```diff
@@ -0,0 +1,42 @@
+require 'test_helper'
+
+class UserTest < ActiveSupport::TestCase
+  def setup
+    @github_omniauth_hash = OmniAuth.config.mock_auth[:github]
+    @user                 = users(:jim)
+  end
+
+  context 'validations' do
+    context 'presence' do
+      [:provider, :uid, :login, :email, :token].each do |column|
+        should validate_presence_of(column)
+      end
+    end
+
+    context 'uniqueness' do
+      [:uid, :login, :email, :token].each do |column|
+        should validate_uniqueness_of(column)
+      end
+    end
+  end
+
+  test 'create a new valid user from the auth hash' do
```

Comment:
When I use Rspec I usually have a `describe` block to group all my tests of a particular method. You might consider using a context to do the same here.

```ruby
context '#create_from_auth_hash' do
  test 'creates valid user' do
    assert_difference 'User.count', 1 do
    User.create_from_auth_hash(@github_omniauth_hash)
  end
end

This is the style I'm most used to, again up to you how you want to do it.